BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -134,19 +134,23 @@ extension ElementContent.Builder: ContentStorage {
 
         var result: [(identifier: ElementIdentifier, node: LayoutResultNode)] = []
         result.reserveCapacity(children.count)
+        
+        let identifierFactory = ElementIdentifier.Factory(reserveCapacity: children.count)
 
         for index in 0..<children.count {
-            let currentChildLayoutAttributes = childAttributes[childAttributes.startIndex.advanced(by: index)]
-            let currentChild = children[children.startIndex.advanced(by: index)]
+            let currentChildLayoutAttributes = childAttributes[index]
+            let currentChild = children[index]
 
             let resultNode = LayoutResultNode(
                 element: currentChild.element,
                 layoutAttributes: currentChildLayoutAttributes,
-                content: currentChild.content)
-
-            let identifier = ElementIdentifier(
-                key: currentChild.key,
-                index: index)
+                content: currentChild.content
+            )
+            
+            let identifier = identifierFactory.identifier(
```

Comment:
For readers: This is where the valuable change which actually improves view diffing happens.